### ISSUE TITLE
Select files: improve exclusion UX

### ISSUE THREAD

**Issue by kompot** [t9md/atom-narrow#173]
Hi,
a proposal for probable select files UX improvement.

If I'm right, right now the only way to reduce number of files matched is by inputting some glob-like pattern `aaa bbb ccc!` and then confirm by pressing enter. Is this the only way to reduce matched files?

If that's the only way I've a proposal that might be a natural fit for `narrow` (actually I thought that it should "just work" but it didn't).

1. Select files is just plain buffer as far as I understand.
2. Any line that is deleted from it - should be excluded.
3. Any valid editor combination could be used. Like manually selecting some line and deleting it with backspace.
4. Press enter to confirm.

So the pattern is the following - file list is filtered by glob entered and then those entries that were manually deleted are excluded from that list.

**Comment by t9md** [comment]
> If I'm right, right now the only way to reduce number of files matched is by inputting some glob-like pattern aaa bbb ccc! and then confirm by pressing enter. Is this the only way to reduce matched files?

You can also exclude specific file by `backspace` on item(not in select-file UI, in normal provider UI like result of `narrow:search`).

I know your point, but I want select-file preference is remembered and applied on next narrow invocation.
I can implement your enhancement request, but I imagine it also break current usability for remember select-files query and apply on startup etc..
DIdn't investigated yet.

**Comment by t9md** [comment]
General filtering should be done not by select-file, by normal narrow-query.
select-file should be just a helper.

Maybe the project you are working on is different than mine.
If you give me example project(repo) and search example, and show how tedious to exclude unwanted item, I can take your suggestion more seriously.

**Comment by kompot** [comment]
Well, mostly I wanted to be able to exclude directories like described in
https://github.com/t9md/atom-narrow/issues/172

That's the main point. And adding `|` to `Query` would solve that.

But I thought that removing items with `backspace` both on normal provider UI and in selected files might look convenient and feel natural. Just feels strange that you can exclude with backspace from normal provider UI and can't exclude from select files (without remembering between invocations of course).

**Comment by t9md** [comment]
understand your point, but things is not easy, maybe because of implementation, and I prioritize internal consistency over usefulness of user.

I want select-files is **reproduceable** by query on select-file. When I introduce this special select-file provider, I tried and evaluated different idea but most of it was removed.
Now what I think it's important is **reproducability** by select-file query.
When I added `backspace` deletion on select-file, I also want to it reflect it as `query`, but adding long file name on query like `!/very/long/file/path/lib/hoge.js` is looks ugly and reduce usability of query input.

The `backspace` on item on normal provider exclude files and it's not reflected as select-file query. I know it, it's already inconsistent.

Anyway, I'll consider `|` query expression, it's generally useful, I sometimes hit situation where I want to do OR search(but currently can't)
